src/methods/michelsen/saturation_pure.jl

```diff
@@ -205,6 +205,22 @@ function psat(model::EoSModel, T; p0=nothing, vol0=(nothing, nothing))
     return P, vol_liq, vol_vap
 end
 
+struct ChemPotDensitySaturation{T} <: SaturationMethod
```

Comment:
I only put everything together as a separate method but probably there is some optimizations here to be done

Reply:
we have `x0_sat_pure(model,T)`, that can create good initial volumes at a specific temperature, and some methods have specializations (IAPWS95 has a specific water correlation, for example). i will plugin that in the case that both vv and vl are `nothing`